app/controllers/repositories_controller.rb

```diff
@@ -27,16 +27,33 @@ def index
   end
 
   def destroy
-    super
-  rescue Ontology::DeleteError => e
+    resource.destroy_asynchronously
+    redirect_to repositories_path
+  rescue Repository::DeleteError => e
     flash[:error] = e.message
     redirect_to resource
   end
 
+  def undestroy
+    resource.undestroy
+    begin
+      resource.reload
+      flash[:success] = t('repository.undestroy.success')
+      redirect_to resource
+    rescue ActiveRecord::RecordNotFound
+      flash[:alert] = t('repository.undestroy.failure')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.